### PR TITLE
TDI-36303: Change  RuntimeException to ComponentException.

### PR DESCRIPTION
**Pull request by jzhao-talend**
https://jira.talendforge.org/browse/TDI-36303

### CODE REVIEW

File: components-salesforce/src/main/java/org/talend/components/salesforce/runtime/SalesforceAvroRegistry.java [hunk 1]
```diff
@@ -284,7 +285,7 @@ public Long convertToAvro(String value) {
             } catch (ParseException e) {
                 // TODO: error handling
                 e.printStackTrace();
```
Comment (sgandon) [inline components-salesforce/src/main/java/org/talend/components/salesforce/runtime/SalesforceAvroRegistry.java]: no printStackTrace() ever please.

File: components-salesforce/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java [hunk 2]
```diff
@@ -454,7 +455,7 @@ else if (TSalesforceOutputProperties.FIELD_ERROR_FIELDS.equals(outField.name()))
             String ID = "Id";
             Schema.Field idField = input.getSchema().getField(ID);
             if (null == idField)
-                throw new RuntimeException(ID + " not found");
+                throw new ComponentException(new Throwable(ID + " not found"));
```
Comment (sgandon) [inline components-salesforce/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java]: please suround this with brackets {}

File: components-salesforce/src/main/java/org/talend/components/salesforce/runtime/SalesforceBulkRuntime.java [hunk 3]
```diff
@@ -421,7 +396,7 @@ public void doBulkQuery(String moduleName, String queryStatement, int secToWait)
                 queryResultIDs = new HashSet<String>(Arrays.asList(list.getResult())).iterator();
                 break;
             } else if (info.getState() == BatchStateEnum.Failed) {
-                throw new RuntimeException("-------------- failed ----------" + info);
+                throw new ComponentException(new Throwable("-------------- failed ----------" + info));
```
Comment (sgandon) [inline components-salesforce/src/main/java/org/talend/components/salesforce/runtime/SalesforceBulkRuntime.java]: I don't like the fact you create a Throwable for some expected error. I would advise you to create one specific ErrorCode for this case and use it with a context, see this for example.
https://github.com/Talend/components/blob/master/components-common/src/main/java/org/talend/components/common/FixedConnectorsComponentProperties.java#L63
I am working at making it more simpler to setup context but meanwhile you can do like the example.

File: components-salesforce/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java (hunk 2 above)
Comment (sgandon) [inline components-salesforce/src/main/java/org/talend/components/salesforce/runtime/SalesforceWriter.java]: same comment as above, please create some specific ErrorCode.

File: components-salesforce/src/main/java/org/talend/components/salesforce/runtime/SalesforceBulkRuntime.java (hunk 3 above)
Comment (jzhao-talend) [inline components-salesforce/src/main/java/org/talend/components/salesforce/runtime/SalesforceBulkRuntime.java]: @sgandon 
Thanks for your checking.
So I need to create enum like this?
`public enum SalesforceRuntimeErrorCode implements ErrorCode {
    EXECUTE_BATCH_FAILED(HttpServletResponse.SC_BAD_REQUEST, "runtime"), 
    SF_ID_MISSING(HttpServletResponse.SC_BAD_REQUEST, "runtime");
//....
}`

**Pull request merged**